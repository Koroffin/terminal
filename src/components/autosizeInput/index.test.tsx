import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { adjustInput, AutosizeInput } from './index';

describe('AutosizeInput', () => {
  it('should render correctly', () => {
    const { container } = render(<AutosizeInput />);
    expect(container).toMatchSnapshot();
  });
  it('should call onChange when input is changed', () => {
    const onChange = jest.fn();
    const { getByTestId } = render(<AutosizeInput onChange={onChange} />);
    const input = getByTestId('autosize-input');
    fireEvent.change(input, { target: { value: 'test' } });
    expect(onChange).toHaveBeenCalled();
  });
  it('should call adjustInput when input is changed', () => {
    const { getByTestId } = render(<AutosizeInput />);
    const input = getByTestId('autosize-input');
    fireEvent.change(input, { target: { value: 'test' } });
    expect(input.style.width).toBe('4ch');
  });
  it('should not call adjustInput if input is not defined', () => {
    expect(adjustInput(undefined)).toBeUndefined();
  });
  it('should call onInput when input is changed', () => {
    const onInput = jest.fn();
    const { getByTestId } = render(<AutosizeInput onInput={onInput} />);
    const input = getByTestId('autosize-input');
    fireEvent.input(input, { target: { value: 'test' } });
    expect(onInput).toHaveBeenCalled();
  });
});
