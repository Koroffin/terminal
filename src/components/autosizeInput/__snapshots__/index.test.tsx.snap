// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['AutosizeInput should render correctly 1'] = `
<div>
  <input
    class="sc-bczRLJ ZLNSN"
    data-autotest-id="autosize-input"
  />
</div>
`;
